### PR TITLE
Creation of filter server

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Implement persistence with Postgres sql

File: README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[![Actions Status](https://github.com/gridsuite/filters-server/workflows/CI/badge.svg)](https://github.com/gridsuite/filters-server/actions)
```
Comment (FranckLecuyer) [inline README.md:1]: filter-server

File: README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+[![Coverage Status](https://sonarcloud.io/api/project_badges/measure?project=org.gridsuite%3Afilters-server&metric=coverage)](https://sonarcloud.io/component_measures?id=org.gridsuite%3Afilters-server&metric=coverage)
+[![MPL-2.0 License](https://img.shields.io/badge/license-MPL_2.0-blue.svg)](https://www.mozilla.org/en-US/MPL/2.0/)
+# filters-server
```
Comment (FranckLecuyer) [inline README.md:4]: filter-server

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,206 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline pom.xml:4]: 2021

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+-->
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-parent-ws</artifactId>
+        <version>3</version>
```
Comment (FranckLecuyer) [inline pom.xml:18]: 4

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        <relativePath/>
+    </parent>
+
+    <groupId>org.gridsuite</groupId>
+    <artifactId>gridsuite-filters-server</artifactId>
```
Comment (FranckLecuyer) [inline pom.xml:23]: gridsuite-filter-server

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+            <email>jacques.borsenberger@rte-france.com</email>
+            <organization>RTE</organization>
+            <organizationUrl>http://www.rte-france.com</organizationUrl>
+        </developer>
+    </developers>
+
+    <properties>
+        <java.version>11</java.version>
+
+        <gridsuite-dependencies.version>4</gridsuite-dependencies.version>
```
Comment (FranckLecuyer) [inline pom.xml:49]: 6

File: src/main/java/org/gridsuite/filters/server/FilterApi.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filters/server/FilterApi.java:2]: 2021

File: src/main/java/org/gridsuite/filters/server/FilterApplication.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filters/server/FilterApplication.java:2]: 2021

File: src/main/java/org/gridsuite/filters/server/FilterController.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filters/server/FilterController.java:2]: 2021

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Persistence with Postgres sql instead of cassandra
Remove all dependencies to cassandra

File: src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
+package org.gridsuite.filter.server.entities;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java:1]: Add license and author

File: src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    @ElementCollection
+    @CollectionTable(foreignKey = @ForeignKey(name = "lineFilterEntity_countries_fk2"), indexes = {@Index(name = "lineFilterEntity_countries_idx2", columnList = "lineFilterEntity_name")})
+    Set<String> countries2;
+
+    @Column(name = "substationName1")
+    String substationName1;
+    @Column(name = "substationName2")
+    String substationName2;
+
+    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java:31]: avoid using EAGER (LAZY instead)

File: src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    @JoinColumn(name  =  "numericFilterId1_id",
+        referencedColumnName  =  "id",
+        foreignKey = @ForeignKey(
+            name = "numericFilterId_id_fk1"
+        ), nullable = true)
+    NumericFilterEntity nominalVoltage1;
+
+    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java:39]: avoid using EAGER (LAZY instead)

File: src/main/java/org/gridsuite/filter/server/entities/AbstractFilterEntity.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/entities/AbstractFilterEntity.java:2]: 2021

File: src/main/java/org/gridsuite/filter/server/dto/ScriptFilter.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/dto/ScriptFilter.java:2]: 2021

File: src/main/java/org/gridsuite/filter/server/dto/NumericalFilter.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
+package org.gridsuite.filter.server.dto;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/dto/NumericalFilter.java:1]: Add license and author

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 16]
```diff
@@ -0,0 +1,243 @@
...
+        }
+        return false;
+    }
+}
+
+@Service
+public class FilterService {
+
+    @Autowired
+    FilterService self;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:91]: Here, self is used to auto inject the service for internal @Transactional method calls support
You need to annotate some methods with @Transactional (if only to handle the lazy collections in LineFilterEntity), and calling these methods In Spring cannot be done from the same service instance.

See ContingencyListService in actions-server, or StudyService in study-server

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java (hunk 11 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java:31]: what use is lazy if we always fetch it when we get the filter ?

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 16 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:91]: it should be for the rename

Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:91]: and it is already on insert

File: src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java (hunk 11 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java:31]: In getFiltersNames() method, we don't need to fetch nominalVoltage1 and nominalVoltage2 field. 
It's the same for countries1 and countries2 field, and these fields are handled lazily by default.

Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/filter/server/entities/LineFilterEntity.java:31]: i've forgotten to write the custom query for getFiltersNames(), it should be added in my next push

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 16 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:87]: You can remove 'self' field in FilterService, because you don't use it ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
There are still 2 of my conversations not resolved ... (in README.md and FilterService)

**Review by niconoir — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**